sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamBuiltinAggregations.java

```diff
@@ -453,4 +462,57 @@ public Long extractOutput(Accum accum) {
       return accum.bitAnd;
     }
   }
+
+  public static class LogicalAnd extends CombineFn<Boolean, LogicalAnd.Accum, Boolean> {
+    static class Accum {
+      // Initially, input is empty
+      boolean isEmpty = true;
+      // true if any null value is seen in the input, null values are to be ignored
+      boolean isNull = false;
+      // logical_and operation result
+      boolean logicalAnd = Boolean.parseBoolean(null);
+    }
+
+    @Override
+    public Accum createAccumulator() {
+      return new Accum();
+    }
+
+    @Override
+    public Accum addInput(Accum accum, Boolean input) {
+      if (input == null) {
+        accum.isNull = true;
+        accum.isEmpty = false;
+        return accum;
+      }
+      accum.isEmpty = false;
+      accum.logicalAnd = accum.logicalAnd && input;
+      return accum;
+    }
+
+    @Override
+    public Accum mergeAccumulators(Iterable<Accum> accums) {
+      LogicalAnd.Accum merged = createAccumulator();
+      for (LogicalAnd.Accum accum : accums) {
+        if (accum.isNull) {
+          // ignore null case
+          continue;
```

Comment:
`testLogicalAndWithNullAndNoNullInput` is failing because this logic is incorrect. This logic looks to be partially copied from `BitAnd`, which handles nulls completely differently. In `BitAnd`, we return null if _any_ values were null; in `LogicalAnd`, we want to return null only if _all_ values were null.

To fix this, you will need to fix both `addInput` and `mergeAccumulators` to properly handle nulls.

Reply:
Hi @sonam-vend, please see the comment above to fix the problem.

Reply:
@ibzib @robinyqiu I consider there is a difference between logical_And and bit_And implementation when it comes to handling null values.  What I am actually trying to do is to ignore all the null values in the input. For instance,
Input: [true, null, false]
I want to ignore null value in the input and perform logical_And and for [true, false].